src/clr-angular/utils/focus/key-focus/key-focus.module.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+
+import { KEY_FOCUS_DIRECTIVES } from './index';
```

Comment:
It's actually defined there, in the index.ts file. It can't be imported from elsewhere, so it can't lead to cyclic dependencies.
This is a pattern we've been using for smaller modules all over the project and especially in the utils: popover module, conditional module, focus trap, outside-click, etc.

Reply:
We may start moving these definitions to the modules if you think it will be better for them to be there?

Reply:
I'm not sure what the right action is. we don't have a preference written down in the CODING_GUIDELINES. 

I was thinking of importing the entity when using it in code when I wrote that but since its defined there and consumed in the module it should be ok as it is.

Reply:
There was a time I moved all of the definitions out of the index files. We got bit several times with compiler errors from this, so I moved them all to other places. Maybe the smaller modules got missed or weren't causing issues, but if we aren't publicly exporting these we don't even need an index. I suspect that is why there are some still in there, since I worked backwards from the public_api file to find and refactor all index files.

Reply:
I don't think the TSLint rule will catch this on a non-publicly exported item. We should still just have one pattern, which is nothing should be defined in an index file. I did move them to the corresponding modules during my refactor.